src/extension.ts

```diff
@@ -0,0 +1,140 @@
+import { from } from 'rxjs'
+import { filter, switchMap } from 'rxjs/operators'
+import * as sourcegraph from 'sourcegraph'
+import { resolveSettings, Settings } from './settings'
+
+/**
+ * Params derived from the document's URI.
+ */
+interface Params {
+    repo: string | null
+    file: string | null
+}
+
+interface SentryProject {
+    projectId: string
+    lineMatches: RegExp[]
+}
+
+const DECORATION_TYPE = sourcegraph.app.createDecorationType()
+const SETTINGSCONFIG = resolveSettings(sourcegraph.configuration.get<Settings>().value)
+
+export function activate(context: sourcegraph.ExtensionContext): void {
+    sourcegraph.workspace.onDidOpenTextDocument.subscribe(textDocument => {
+        const params: Params = getParamsFromUriPath(textDocument.uri)
+        const sentryProjects = SETTINGSCONFIG['sentry.projects']
+
+        // Retrieve the Sentry project that this document reports to.
+        const sentryProject = matchSentryProject(params, sentryProjects)
+
+        if (sourcegraph.app.activeWindowChanges && sentryProjects && sentryProject) {
+            const activeEditor = from(sourcegraph.app.activeWindowChanges).pipe(
+                filter((window): window is sourcegraph.Window => window !== undefined),
+                switchMap(window => window.activeViewComponentChanges),
+                filter((editor): editor is sourcegraph.CodeEditor => editor !== undefined)
+            )
+
+            // When the active editor changes, publish new decorations.
+            context.subscriptions.add(
+                activeEditor.subscribe(editor => {
+                    decorateEditor(editor, sentryProject.projectId, sentryProject.lineMatches)
+                })
+            )
+        }
+    })
+}
+
+/**
+ * Extract Sentry params from document URI necessary to
+ * check if the current document sends log events to Sentry and
+ * use these params build URL to the Sentry issues stream page.
+ * @param textDocument A document URI.
+ * @returns repo and file part of URI.
+ */
+export function getParamsFromUriPath(textDocument: string): Params {
+    // TODO: Support more than just GitHub.
+    const repoPattern = /github\.com\/([^\?\#]+)/gi
+    const filePattern = /#([^\?\#\/]+)\/.*\.?$/gi
+    const repoM = repoPattern.exec(textDocument)
+    const fileM = filePattern.exec(textDocument)
+    return {
+        repo: repoM![1],
```

Comment:
Please do not use the `!` operator here. `exec()` may return null, and this code will throw a `TypeError` in that case.

Reply:
Yes, I want to work on a separate PR that will take care of all kinds of exceptions such as this one.